api_app/script_analyzers/file_analyzers/triage_scan.py

```diff
@@ -0,0 +1,89 @@
+
+class TriageScanFile(classes.FileAnalyzer):
+    # using public endpoint as the default url
+    base_url: str = "https://api.tria.ge/v0/"
+    private_url: str = "https://private.tria.ge/api/v0/"
+
+    def set_config(self, additional_config_params):
+        self.endpoint = additional_config_params.get("endpoint", "public")
+        if self.endpoint == "private":
+            self.base_url = self.private_url
+
+        self.api_key_name = additional_config_params.get("api_key_name", "TRIAGE_KEY")
+        self.__api_key = secrets.get_secret(self.api_key_name)
+        self.max_tries = additional_config_params.get("max_tries", 200)
+        self.poll_distance = 3
+
+    def run(self):
+        final_report = {}
+        if not self.__api_key:
+            raise AnalyzerRunException(
+                f"No API key retrieved with name: {self.api_key_name}"
+            )
+
+        headers = {"Authorization": f"Bearer {self.__api_key}"}
+
+        name_to_send = self.filename if self.filename else self.md5
+        binary = get_binary(self.job_id)
+        files = {
+            "file": (name_to_send, binary),
+            "_json": (None, b'{"kind": "file", "interactive": false}'),
+        }
+
+        logger.info(f"triage md5 {self.md5} sending sample for analysis")
+        response = requests.post(
+            self.base_url + "samples", headers=headers, files=files
+        )
+        sample_id = response.json()["id"]
+
+        for _try in range(self.max_tries):
+            if response.status_code != 200:
+                time.sleep(self.poll_distance)
+                logger.info(f"triage md5 {self.md5} polling for result try #{_try + 1}")
+                result_url = response.json().get("result_url", "")
+                response = requests.get(self.base_url + result_url)
+                response.raise_for_status()
+
+        if response.status_code != 200:
+            raise AnalyzerRunException("max retry attempts exceeded")
+
+        # Event stream is opened. Updates till the task is completed
+        requests.get(self.base_url + f"samples/{sample_id}/events", headers=headers)
+
+        # Get short summary of sample
+        summary = requests.get(
+            self.base_url + f"samples/{sample_id}/summary",
+            headers=headers,
+        )
+        summary_json = summary.json()
+        final_report["summary"] = summary_json
+
+        # Get overview report of sample
+        overview = requests.get(
+            self.base_url + f"samples/{sample_id}/overview.json",
+            headers=headers,
+        )
+        overview_json = overview.json()
+        final_report["overview"] = overview_json
+
+        # Get static report of sample
+        static_report = requests.get(
+            self.base_url + f"samples/{sample_id}/reports/static",
+            headers=headers,
+        )
+        static_report_json = static_report.json()
+        final_report["static_report"] = static_report_json
+
+        return final_report
```

Comment:
could you please provide screenshot or paste of some example reports?

Reply:
I ran the test on a random sample i took from the publicly available ones. Summary and the other reports might have some redundant stuff. I found these endpoints in their docs and just compiled all the results. Will change/ remove things not needed.
[Tree format from the GUI](https://imgur.com/a/BvdtVgX)
[Raw](https://imgur.com/a/ln0XstJ)

Reply:
Yes, I was thinking the same. My concern is to avoid duplicated data in the result to avoid confusion.

At a first glance it seems that the "summary" data are replicated in the "overview". So we can remove it I think.

What about the data in the "static" report? are they replicated in the "overview" too?

Reply:
I removed summary and added task specific reports. I noticed on the wesbite, that the analyzed report contains 4 or more parts : `overview`, `static_report`, `<task_1>`, `<task_2>` and so on as seen below. So i added task wise report alongside overview and static in the final json. All of them share some basic common info but have their own unique analysis so I have kept all of them and removed only summary.

![image](https://user-images.githubusercontent.com/17424778/99437463-8028d880-2938-11eb-9e10-414c1abc76a1.png)

Reply:
ok, thank you for explaining this! 

Anyway I would change this to avoid too much data in the report. The best option could be to make this configurable. For instance, by default this analyzer could provide only the "overview" report. And, with a specific parameter, the analysis report could be "complete" with the static and the tasks reports too.

Reply:
Added an option in `additional_config_params`  for this.